### PR TITLE
Add CSV support to plotBuildData

### PR DESCRIPTION
**Pull request by langers2k**

### CODE REVIEW

**Comment by langers2k** [comment]
I'll add doc's if that looks a sensible was of dealing with inclusion flags.

**Comment by daspilker** [comment]
Hm, PlotXMLSeriesContext does not use an enum for the node types, so PlotCSVSeriesContext should not use an enum either. And depending on the inclusion flag, addiontial XML element like colExclusionSet and strExclusionSet must be generated.

How about adding dedicated methods instead of the enum?

```groovy
includeColumns(String... columnNames)
excludeColumns(String... columnNames)
includeColumns(int.. columIndexes)
excludeColumns(int.. columIndexes)
```

**Comment by langers2k** [comment]
Hadn't noticed the two additional XML elements.

I'd rather keep the string than expect the user to add each index on a separate line. This is because selecting via indexes the plugin still expects a string of comma delimited values and when migrating to job-dsl it's just copy/paste.

Happy with the idea of dedicated methods but I'm thinking:
```groovy
includeByName(String criteria)
excludeByName(String criteria)
includeByIndex(String criteria)
excludeByIndex(String criteria)
```

Each of which will set `String inclusionFlag` and  either `String colExclusionSet` or `String strExclusionSet`.

**Comment by langers2k** [comment]
Removed the enum and moved to dedicated methods with the elements that I'd missed previously

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy [hunk 1]
```diff
@@ -390,6 +397,22 @@ class PublisherContext implements Context {
...
+                                        inclusionFlag(data.inclusionFlag)
+                                        if ((data.inclusionFlag == 'INCLUDE_BY_STRING') ||
+                                            (data.inclusionFlag == 'EXCLUDE_BY_STRING')) {
+                                            strExclusionSet {
+                                                string(data.exclusionValues ?: '')
+                                            }
+                                        }
+                                        if ((data.inclusionFlag == 'INCLUDE_BY_COLUMN') ||
+                                            (data.inclusionFlag == 'EXCLUDE_BY_COLUMN')) {
+                                            colExclusionSet()
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy]: There is something missing. The indexes need to be integers and then the `colExclusionSet` is filled. That's why I suggested to use int arguments in this case...

```xml
<hudson.plugins.plot.CSVSeries>
  <file>test</file>
  <label/>
  <fileType>csv</fileType>
  <colExclusionSet>
    <int>1</int>
    <int>2</int>
    <int>6</int>
  </colExclusionSet>
  <inclusionFlag>EXCLUDE_BY_COLUMN</inclusionFlag>
  <exclusionValues>1,2,6</exclusionValues>
  <url>bar</url>
  <displayTableFlag>true</displayTableFlag>
</hudson.plugins.plot.CSVSeries>
```

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy [hunk 2]
```diff
@@ -350,7 +358,6 @@ class PublisherContext implements Context {
      *                 </hudson.plugins.plot.PropertiesSeries>
      *                 <hudson.plugins.plot.XMLSeries>
      *                     <file>XML</file>
-     *                     <label></label>
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/publisher/PublisherContext.groovy]: The label element is generated for all types of series. Lame, but we should emulate that to keep the diff small.

**Comment by daspilker** [comment]
There are merge conflicts because I removed all XML examples. Please rebase your commits to master.

**Comment by langers2k** [comment]
Rebased and flattened.

- Added label to xml files
- Corrected usage of colExclusionSet and strExclusionSet
- Moved to suggestion for adding inclusion/exclusion sets
- Added checks to ensure that names/indexes aren't mixed up
- Added checks to ensure that includes/excludes aren't mixed up

**Comment by daspilker** [comment]
Great. Update the docs and I will take care of the rest.

**Pull request merged**